src/Integration/MefServices/VsSessionHost.cs

```diff
@@ -209,10 +209,10 @@ private void OnActiveSectionChanged()
         private void OnActiveSolutionChanged(object sender, ActiveSolutionTrackerEventArgs args)
         {
             // Reset, and abort workflows
-            this.ResetBinding(abortCurrentlyRunningWorklows: true);
+            this.ResetBinding(abortCurrentlyRunningWorklows: true, clearCurrentBinding: !args.IsSolutionOpen);
         }
 
-        private void ResetBinding(bool abortCurrentlyRunningWorklows)
+        private void ResetBinding(bool abortCurrentlyRunningWorklows, bool clearCurrentBinding)
```

Comment:
Both the _ActivSolutionBoundTracker_ and the session host are listening for the solution close event and we don't know which order they will be called in. This is a problem because the session host reads the binding configuration, which may or may have been reset to standalone by the ASBT.
This flag has been added as a workaround to force the session host to clear the binding when the solution is closing.

A better fix would be to simplifying the eventing model, but that's too big a change for now.

Reply:
Let's add the TODO directly in the code, WDYT?